src/vs/platform/diagnostics/node/diagnosticsService.ts

```diff
@@ -61,12 +66,146 @@ export async function collectWorkspaceStats(folder: string, filter: string[]): P
 		{ tag: 'dockerfile', filePattern: /^(dockerfile|docker\-compose\.ya?ml)$/i }
 	];
 
+	let rootFileMatchers: RootFileMatcher[];
+
+	// Linux is omitted because few cloud sync clients support it, and for those who are available on Linux, there are multiple clients and they can be configured differently
+	switch (process.platform) {
+		case 'win32':
+			const homeDir = osLib.homedir().toLowerCase();
+			rootFileMatchers = [
+				{
+					tag: 'gdrive', matcher: (path) => {
+						path = path.toLowerCase();
+						// File Streaming mode
+						if (path.match(/^[a-z]:\\(my drive|shared drives)\\/)) {
+							return true;
+						}
+						// Mirror Files mode
+						if (path.startsWith(homeDir + '\\my drive\\')) {
+							return true;
+						}
+						return false;
+					}
+				},
+				{
+					tag: 'dropbox', matcher: (path) => {
+						return path
+							.toLowerCase()
+							.startsWith(homeDir + '\\dropbox'); // Ending in *
+					}
+				},
+				{
+					tag: 'onedrive', matcher: (path) => {
+						return path
+							.toLowerCase()
+							.startsWith(homeDir + '\\onedrive'); // Ending in *
+					}
+				},
+				{
+					tag: 'box', matcher: (path) => {
+						return path
+							.toLowerCase()
+							.startsWith(homeDir + '\\box\\');
+					}
+				},
+				{
+					tag: 'nextcloud', matcher: (path) => {
+						return path
+							.toLowerCase()
+							.startsWith(homeDir + '\\nextcloud\\');
+					}
+				},
+				{
+					tag: 'owncloud', matcher: (path) => {
+						return path
+							.toLowerCase()
+							.startsWith(homeDir + '\\owncloud\\');
+					}
+				},
+			];
+			break;
+
+		case 'darwin':
+			rootFileMatchers = [
+				{
+					tag: 'gdrive', matcher: (path) => {
+						path = path.toLowerCase();
+						// File Streaming mode
+						if (path.startsWith('/volumes/googledrive/')) {
```

Comment:
can't these be simplified to `return path.startsWith('/volumes/googledrive/') || path.startsWith(homeDir + '/my drive/');`